### PR TITLE
Add pipeline blend modes & demo

### PR DESCRIPTION
**Pull request by bdero**
Part of https://github.com/flutter/flutter/issues/99520
Depends on https://github.com/flutter/impeller/pull/57

* Use premultiplied source colors to allow for symmetrical alphablend ops. I.e. `SourceOver` variations can be done without premultiplied sources, but `DestinationOver` variations cannot.
* All our textures are unpremultiplied right now, so I added a conversion for  the fragment stage. Might revisit while working on [this](https://github.com/flutter/flutter/issues/99521).
* Will add [more blends](https://api.skia.org/SkBlendMode_8h.html#ad96d76accb8ff5f3eafa29b91f7a25f0) and extend the test in a follow up.

https://user-images.githubusercontent.com/919017/156912575-359c5c43-871a-40f2-a227-88e2741e898a.mov

### CODE REVIEW

**Comment by bdero** [comment]
Hmm, pretty sure I've got these wrong at the moment

File: entity/entity.h [hunk 1]
```diff
@@ -18,6 +18,17 @@ class RenderPass;
 
 class Entity {
  public:
+  /// All pipeline blend mode presets assume that both the source (fragment
+  /// output) and destination (first color attachment) have colors with
+  /// premultiplied alpha.
+  enum class PipelineBlendMode {
```
Comment (chinmaygarde) [inline entity/entity.h:24]: Nit: Just `BlendMode` should be fine. That it is affected using a pipeline variant is of no concern to the user of the entity framework.

File: entity/contents/solid_stroke_contents.cc [hunk 2]
```diff
@@ -143,7 +143,7 @@ bool SolidStrokeContents::Render(const ContentContext& renderer,
                    entity.GetTransformation();
 
   VS::StrokeInfo stroke_info;
-  stroke_info.color = color_;
+  stroke_info.color = color_.Premultiply();
```
Comment (chinmaygarde) [inline entity/contents/solid_stroke_contents.cc:146]: I think this is an orthogonal concern. One that should be handled by the users of the entity framework. WDYT?

Comment (chinmaygarde) [inline entity/contents/solid_stroke_contents.cc:146]: I think this is an orthogonal concern. One that should be handled by the users of the entity framework. WDYT?

**Review by chinmaygarde — COMMENTED** [review]
One nit and one comment about concerns about whether the colors are premultiplied or not being orthognal.

File: entity/contents/solid_stroke_contents.cc (hunk 2 above)
Comment (bdero) [reply entity/contents/solid_stroke_contents.cc:146]: Sure, I think it makes sense to bubble up this concern.

File: entity/entity.h (hunk 1 above)
Comment (bdero) [reply entity/entity.h]: Done.

File: entity/contents/solid_stroke_contents.cc (hunk 2 above)
Comment (bdero) [reply entity/contents/solid_stroke_contents.cc]: Done.

**Review by chinmaygarde — APPROVED** [review]
